client/landing/stepper/declarative-flow/internals/hooks/use-step-navigation-with-tracking/index.ts

```diff
@@ -54,11 +54,13 @@ export const useStepNavigationWithTracking = ( {
 		() => ( {
 			...( stepNavigation.submit && {
 				submit: ( providedDependencies: ProvidedDependencies = {}, ...params: string[] ) => {
```

Comment:
This event triggers every time you call the submit function, which is why we're seeing calypso_signup_actions_submit_step event triggered multiple times within the use-my-domain component